src/common/history.lisp

```diff
@@ -57,30 +57,42 @@
     (aref (history-data history)
           (1- (length (history-data history))))))
 
-(defun add-history (history input &key (allow-duplicates t) (test #'equal))
-  "Add this input to the history.
-
-  Don't add the same input as the previous one.
-  If allow-duplicates is non t, don't add duplicates at all.
-  If limit is set, overwrite oldest entry when reached."
-  (when (or allow-duplicates
-            (not (find input (history-data history) :test test)))
-    (when (require-additions-to-history-p input (last-history history))
-      (let ((limit (history-limit history)))
-        (cond
-          ((and limit (>= (length (history-data history)) limit))
-           ;; Shift by 1, overwriting the oldest
-           (replace (history-data history)
-                    (history-data history)
-                    :start1 0 
-                    :start2 1
-                    :end2 limit)
-           (setf (aref (history-data history) (1- limit)) input))
-          (t
-           ;; Add new element normally
-           (vector-push-extend input (history-data history)))))
-      (setf (history-index history)
-            (length (history-data history)))))
+(defun add-history (history input &key (allow-duplicates t) (test #'equal) move-to-top)
```

Comment:
https://github.com/lem-project/lem/blob/main/tests/common/history.lisp
If possible, I would be happy if a test were added here.

Reply:
Let me know if that works :smile: